### PR TITLE
Refactor interpret module

### PR DESCRIPTION
**Pull request by roshankern**
In this PR, the `interpret_model_coefficients` notebook is refactored to include interpretations for each combination of model type and feature type. The bar graph of the average coefficient value has also been removed as it did not format correctly on some machines and was difficult to interpret.

### CODE REVIEW

File: 4.interpret_model/scripts/nbconverted/interpret_model_coefficients.py [hunk 1]
```diff
@@ -15,216 +16,113 @@
...
+    # load model
+    model = load(model_path)
+    # determine model/feature type from model file name
+    model_type = model_path.name.split("__")[0]
+    feature_type = model_path.name.split("__")[1].replace(".joblib", "")
+
+    print(f"Interpreting model: {model_type} \nTrained with features: {feature_type}")
+
+    # get model coefficients and reshape them into a more useable format
+    coefs = np.abs(model.coef_)
```
Comment (gwaybio) [inline 4.interpret_model/scripts/nbconverted/interpret_model_coefficients.py]: The direction is important, please add the raw coefficients to the table. You can also separately add the absolute value coefficients as well, but in a seperate column

File: 4.interpret_model/scripts/nbconverted/interpret_model_coefficients.py [hunk 2]
```diff
@@ -15,216 +16,113 @@
...
+    coefs = pd.DataFrame(coefs).T
+    coefs.columns = model.classes_
+    print(f"Shape of model coefficients{coefs.shape}")
+
+    # restructure/rename dataframe to tidy long format (see preview below)
+    tidy_data = coefs.stack()
+    tidy_data = pd.DataFrame(tidy_data).reset_index(level=[0, 1])
+    tidy_data.columns = ["Feature_Name", "Phenotypic_Class", "Value"]
```
Comment (gwaybio) [inline 4.interpret_model/scripts/nbconverted/interpret_model_coefficients.py]:
```suggestion
    tidy_data.columns = ["Feature_Name", "Phenotypic_Class", "ML_coefficient"]
```

File: 4.interpret_model/scripts/nbconverted/interpret_model_coefficients.py [hunk 3]
```diff
@@ -15,216 +16,113 @@
...
+
+    # add feature names to coefficientss dataframe
```
Comment (gwaybio) [inline 4.interpret_model/scripts/nbconverted/interpret_model_coefficients.py]:
```suggestion
    # add feature names to coefficients dataframe
```

**Review by gwaybio — APPROVED** [review]

File: 4.interpret_model/scripts/nbconverted/interpret_model_coefficients.py (hunk 1 above)
Comment (roshankern) [reply 4.interpret_model/scripts/nbconverted/interpret_model_coefficients.py]: I ended up going with just the raw coefficients as the absolute value can always be determined later. Added in https://github.com/WayScience/phenotypic_profiling_model/pull/22/commits/d00f94dbaa9c5d36fc43ee0af5432ba72dba46bd

File: 4.interpret_model/scripts/nbconverted/interpret_model_coefficients.py (hunk 3 above)
Comment (roshankern) [reply 4.interpret_model/scripts/nbconverted/interpret_model_coefficients.py]: Added in https://github.com/WayScience/phenotypic_profiling_model/pull/22/commits/d00f94dbaa9c5d36fc43ee0af5432ba72dba46bd

File: 4.interpret_model/scripts/nbconverted/interpret_model_coefficients.py (hunk 2 above)
Comment (roshankern) [reply 4.interpret_model/scripts/nbconverted/interpret_model_coefficients.py]: Added in https://github.com/WayScience/phenotypic_profiling_model/pull/22/commits/d00f94dbaa9c5d36fc43ee0af5432ba72dba46bd

**Pull request merged**